### DISCUSSION TITLE
Profile badges are patronising, ugly and might be used as a signal of superiority

### DISCUSSION THREAD

**Discussion by pond** [community/community#28686] [Other Feature Feedback, Questions, & Ideas]
We're developers, not children; the badges are childish and give very bad impressions in some cases (e.g. the ridiculous "YOLO" for merging a PR without review - what, we're supposed to waste time having humans review PRs from Dependabot which are CI-green, are we?!) or risks being a driver for ivory tower superiority and community tribalism, which is something the developer community already struggles with (I'm not listening to your feature request because I have *badges* and you don't).

If we're going to do these things, can they at least (A) make sense and (B) not look like they're aimed at 5 year olds? It's just embarrassing to have these on my profile. I'd happily have them all gone, forever.

* https://github.com/community/community/discussions/28161
* https://github.com/community/community/discussions/28255

I am not alone.

**EDIT** - I do belatedly note that Profile Settings -> Turn Off Achievements down at the bottom at least lets me turn them all off, which is useful. So long as any of this unprofessional kiddie stuff is in there (I mean, "Gitty Up" - for real?!) then that's certainly how I'll leave things.

**Comment by MythodeaLoL** [comment]
I also think an option to disable this would be nice.

**Reply by radar-blip** [reply]
There is an option to disable, its in your settings. 👍🏼

**Comment by Azayzel** [comment]
> We're developers, not children; the badges are childish and give very bad impressions in some cases (e.g. the ridiculous "YOLO" for merging a PR without review - what, we're supposed to waste time having humans review PRs from Dependabot which are CI-green, are we?!) or risks being a driver for ivory tower superiority and community tribalism, which is something the developer community already struggles with (I'm not listening to your feature request because I have _badges_ and you don't).
> 
> If we're going to do these things, can they at least (A) make sense and (B) not look like they're aimed at 5 year olds? It's just embarrassing to have these on my profile. I'd happily have them all gone, forever.
> 
> * [Style and content of new achievement badges trivialise actual achievements #28161](https://github.com/orgs/community/discussions/28161)
> * [Can i get rid of theses badges? #28255](https://github.com/orgs/community/discussions/28255)
> 
> I am not alone.
> 
> **EDIT** - I do belatedly note that Profile Settings -> Turn Off Achievements down at the bottom at least lets me turn them all off, which is useful. So long as any of this unprofessional kiddie stuff is in there (I mean, "Gitty Up" - for real?!) then that's certainly how I'll leave things.

I hear what you’re saying; however, I find them to be actually quite good. The more you’re working in your codebase, the more you’ll get. And they’re suppose to be light-hearted and just something to chuckle at. 

As developers, our time is cut into slices and each comes with headaches. These are fun and interesting analytics about my git behavior and I’m here for it 🚀

Maybe opting out of the granular features would provide MSFT with more detailed analytics anyways? 🤷‍♂️ 🙏

**Comment by jfullstackdev** [comment]
yes again, there will be opposing views here, as in other threads too

that's why GitHub allowed us to hide those achievements and make your profile private

the only additional feature which is, I think, is good to have is the ability to select the badges you want to display

**Comment by pond** [comment]
Yes, if the feature is to be much use to those of us that think "Gitty Up" or "how quickly I closed a PR" or other such things are dumb and/or should simply never be gamified in any way, then a more granular approach is needed.

I have the following "achievements". Ask yourself: Does the developer community want to encourage the behaviour of aiming for these achievements? And are they actually worthwhile things which are in any way difficult?

* Apparently I've at some point closed a PR within 5 minutes of it being opened, which sounds more like I made a mistake rather than a useful achievement.
* I have merged automatically-generated Dependabot PRs by clicking on the "merge" button.
* I created a repository which has "many stars". Well that sounds good! Except - "many" is only 16, in this case. Oh.
* I have apparently opened pull requests after they were merged. I'm not sure what this means precisely but it certainly also sounds like a mistake, not an achievement.
* Arctic Code Vault, which is the one people seem to generally like.

**Reply by jfullstackdev** [reply]
> * Apparently I've at some point closed a PR within 5 minutes of it being opened, which sounds more like I made a mistake rather than a useful achievement.

in a collaborative scenario, it can mean, all are confident to merge that within five minutes

> * I created a repository which has "many stars". Well that sounds good! Except - "many" is only 16, in this case. Oh.

it's a tier or threshold, the next time you reach the next tier, you will receive another one, take a look at Linus Torvalds, the creator of Linux OS

![image](https://user-images.githubusercontent.com/47092464/183366053-362fcd8d-f8e0-4483-b01b-fe12d4024765.png)

**Reply by pond** [reply]
> in a collaborative scenario, it can mean, all are confident to merge that within five minutes

But there's no way to know that, GitHub certainly doesn't, and there is no way in any degree of even the remotest concept of professionalism that we should be encouraging people to rush PRs to a close within a few minutes, collaborative or not.

> it's a tier or threshold, the next time you reach the next tier, you will receive another one

This is invisible but with tiers surely that just makes it even worse. It's genuinely just gamifying things and encouraging people to do bad things to game a crappy stats engine for asinine rewards. Simultaneously, it's almost _discouraging_ if you're really into this because you have to be _Linus Torvalds_ to get four stars! Unobtainum.

Development, for those that actually enjoy it rather than just do it for a pay cheque, has its own rewards. It has its own aspects of fun, flow, achievement. We don't have to be made to feel like we're role players in a real-life Barney The Dinosaur episode to "**make** it fun".

It's childish and patronising. Aesthetics are subjective, but seriously, an outdated style cat emoji with stars for eyes? It's just a wannabe attempt at kawaii, with all the charm of a middle-age politician trying to convince the electorate that they're "down with the kids".

**Comment by thenithinbalaji** [comment]
Yeah true.
  
1. Achievements should be displayed in a separate tab and not under the profile. 
2. It should be fully transparent, and stuff like PRs and discussions made in private repos should not count for the process. 
3. If someone chose to turn achievements off, then, Github should display the "the user has turned off achievements" message since a few people already believe that more badges = more superior. 
4. Ability to report and remove someone's badge permanently if it's not authentic. Will prevent people from asking others to star their repo or mark theirs as answers in discussions. 
5. Except for Pull Shark and Starstruck badges, the other badges do not support or promote people to contribute to open-source in any way. They must be removed. 
6. Ability to turn off certain badges rather than turning off the entire set of achievements.

If you have any other suggestions, additions, or deletions, add them in the comments. Let's make Github hear!

**Reply by jfullstackdev** [reply]
> 1. Achievements should be displayed in a separate tab and not under the profile.

that will not change anything, what's the difference between placing it in another tab and the option to hide it ? it is still gamification

> 2. It should be fully transparent, and stuff like PRs and discussions made in private repos should not count for the process.

how about those companies that needs to have private repos ? of course they can't make it public

> 3. If someone chose to turn achievements off, then, Github should display the "the user has turned off achievements" message since a few people already believe that more badges = more superior.

still the same thing, it is still gamification

**Reply by thenithinbalaji** [reply]
>that will not change anything, what's the difference between placing it in another tab and the option to hide it? it is still gamification

It would be at least good to have them in a separate tab rather than hiding them completely. Some badges like Arctic Code Vault Contributor, Pull Shark, and Mars Rover Badge, convey your open-source skills but they don't look professional enough to have on the profile page IMO. 

> how about those companies that needs to have private repos ? of course they can't make it public

Added that because I have seen people fake Quick Draw, Pull Shark, Pair extraordinaire, Galaxy Brain, etc. in private repos, and then delete the repo. Even if Github decides to gamify stuff, there must be some level of authenticity right?

---

I am not totally against gamification. I do like badges like Pull Shark, Arctic Code Vault Contributor, and Mars Rover Badge.

**Reply by pond** [reply]
> 1. Achievements should be displayed in a separate tab and not under the profile.

Definitely.

> 2. It should be fully transparent, and stuff like PRs and discussions made in private repos should not count for the process.

Certainly transparent. I realise that people want private repos to be shown but that's just gamifying from behind the barrier of closed source development, _and_ allows people to create fake private repos which with trivial automation could easily provide a fair few of the existing badges for nothing.

You want badges? _Work in open source._ I completely support your idea here.

> 3. If someone chose to turn achievements off, then, Github should display the "the user has turned off achievements" message since a few people already believe that more badges = more superior.

Yeah, "no badges" vs "you don't know because they're hidden" is a useful distinction.

> 4. Ability to report and remove someone's badge permanently if it's not authentic. Will prevent people from asking others to star their repo or mark theirs as answers in discussions.

I'd have to hope that GitHub would have that covered and I'm not sure I like the idea of people policing each other. Unfortunately, people - especially online - are often nasty and petty and could act maliciously in this regard.

> 5. Except for Pull Shark and Starstruck badges, the other badges do not support or promote people to contribute to open-source in any way. They must be removed.

Certainly things that are utterly trivial "achievements" or could represent bad behaviour are very dubious and I have no idea why they were put there, except to say that the entire feature feels like a bunch of corporate suits with not one iota of clue about actual developers, who are pushing something they speak of in terms of "increasing engagement" and "monetisation opportunities".

GitHub does not need to be a social network. Please, goodness, no; we _don't_ need any more. LinkedIn should have been warning enough.

> 6. Ability to turn off certain badges rather than turning off the entire set of achievements.

Absolutely, and surely this would be super, super easy to implement. I'm sure any one of us reading here could add a "hidden" boolean column to the database and implement a "remove" button, with full test coverage, in at most an hour or two.

**Comment by Kordedekehine** [comment]
"...the more you’re working in your codebase, the more you’ll get. And they’re suppose to be light-hearted and just something to chuckle at."-@Azayzel on point

**Comment by Slackadays** [comment]
I don't get how badges are unprofessional. In fact, GH has never claimed to be about professionalism, and actually had the tagline "social coding" for a while.

**Reply by pond** [reply]
If you read the replies above, you'll find they already explain. My TL;DR, noting once again that many constructive suggestions were put forward in the thread above and exactly zero of them seem to have been adopted, is: 

* If the badges didn't look like they were drawn in crayon by a young child, or if they didn't in some cases actively encourage bad practice, I'd have less of an issue with them. But sadly they're IMHO kinda pathetic looking - I don't "chuckle" at something badly drawn with a bad sense of humour behind it, but of course that's 100% subjective - and I *certainly* don't think bad practice should be encouraged at any level; open source or otherwise, professional or otherwise.

> GH has never claimed to be about professionalism

Private GitHub repositories are used by professional organisations for their code. Just because someone in Microsoft marketing put "social coding" on there hardly signals a change of use, nor does it try to do anything other than vaguely hand-wave in the direction of the collaborative nature of Git itself. You will also notice https://github.com/pricing and the "Enterprise" tier. Are you suggesting that's not professional?
